aws/notebooks/hpo.ipynb

```diff
@@ -0,0 +1,1047 @@
+   "metadata": {},
+   "source": [
+    "Based on the dataset size and compute choice we will try to recommend an instance choice, you are of course welcome to select alternate configurations. In the case of the CPU we choose a large memory instance (ml.r5) since the during training we can get upwards of 200GB of memory utilization when using the 10 year dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "instance_type = recommend_instance_type ( code_choice, dataset_directory  ) "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "use_spot_instances_flag = True"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "max_duration_of_experiment_seconds = 60*60*24 # 24 hrs \n",
+    "assert ( max_duration_of_experiment_seconds > 60*60*2 ) # 2 hrs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<span style=\"color:#8735fb; font-size:18pt\"> [ HPO ] </span>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "One of the most important choices when running HPO is to choose the bounds of the hyper-parameter search process. Below we've set the ranges of the hyper-parameters to allow for significant variation in all of the different dimensions and we've selected the Random search strategy. You are of course welcome to modify these settings and experiment."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "search_strategy = 'Random'\n",
+    "assert ( search_strategy in [ 'Random', 'Bayesian' ])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "n_trees_variable_name = 'num_boost_round' if ('XGBoost' in algorithm_choice) else 'n_estimators'\n",
+    "from sagemaker.parameter import ContinuousParameter, IntegerParameter\n",
+    "\n",
+    "hyperparameter_ranges = {\n",
+    "    'max_depth'           : IntegerParameter        ( 5, 15 ),\n",
+    "    n_trees_variable_name : IntegerParameter        ( 100, 500 ),\n",
+    "    'max_features'        : ContinuousParameter     ( 0.1, 1.0 ),\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We also need to decide how may total experiments to run, and how many should run in parallel.\n",
+    "> Note that you may need to request an instance limit increase."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "max_parallel_jobs = 10"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "max_jobs = 100"
```

Comment:
Probably better if default is less than 100, can add a comment for larger jobs

Reply:
Done